cobbler/utils/__init__.py

```diff
@@ -1337,3 +1338,31 @@ def is_str_float(value: str) -> bool:
     except ValueError:
         pass
     return False
+
+
+@contextlib.contextmanager
+def filelock(lock_file: str):
+    """
+    Context manager to acquire a file lock and release it afterwards
+
+    :param lock_file: Path to the file lock to acquire
+    :raises CX: Raised in case of unexpect error acquiring file lock.
+    """
+    fd = os.open(lock_file, os.O_RDWR | os.O_CREAT | os.O_TRUNC, 0o660)
+    lock_file_fd = None
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX)
+        lock_file_fd = fd
+    except Exception as exc:
+        raise CX(
+            f"Something unexpected happen while trying to acquire {lock_file}: {exc}"
+        )
+    finally:
+        if lock_file_fd is None:
+            os.close(fd)
+    try:
+        yield lock_file_fd
+    finally:
+        if lock_file_fd:
+            fcntl.flock(lock_file_fd, fcntl.LOCK_UN)
+            os.close(lock_file_fd)
```

Comment:
How about this simplified version:
```suggestion
    fd = None
    try:
        fd = os.open(lock_file, os.O_RDWR | os.O_CREAT | os.O_TRUNC, 0o660)
        fcntl.flock(fd, fcntl.LOCK_EX)
        yield
    finally:
        if fd:
            with contextlib.suppress(OSError):
                fcntl.flock(fd, fcntl.LOCK_UN)
            os.close(fd)
```
It does not turn one exception into another, but I think that's okay. I assume that unlocking might fail (when locking failed in the first place), which is why I ignored an error for the unlocking. Closing the file descriptor could still fail, but I think it's ok to get the OSError to the caller

Reply:
Looks good to me. I also tested this approach locally and it works fine